source/distributed/tria_base.cc

```diff
@@ -450,7 +450,7 @@ namespace parallel
     // 4) determine the global indices of ghost cells
     std::vector<types::global_dof_index> is_ghost_vector;
     GridTools::exchange_cell_data_to_ghosts<types::global_cell_index>(
-      *this,
+      static_cast<dealii::Triangulation<dim, spacedim> &>(*this),
```

Comment:
This and the other places in this file are, strictly speaking, unrelated. We are calling a function with `parallel::Triangulation` as a template argument. This works, but I had chosen to not make this intermediate class covered by the concept `is_triangulation_or_dof_handler`, rather only selecting the final derived classes as well as `::Triangulation`. That required the change here.

But I think this is useful in its own right because it reduces the number of instantiations of this function with different template arguments.